app/src/main/java/com/gemwallet/android/ui/components/InfoBottomSheet.kt

```diff
@@ -0,0 +1,275 @@
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Delete
+import androidx.compose.material.icons.rounded.Close
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import com.gemwallet.android.localize.R
+import com.gemwallet.android.ui.components.buttons.MainActionButton
+import com.gemwallet.android.ui.components.designsystem.Spacer16
+import com.wallet.core.primitives.TransactionState
+import kotlinx.coroutines.launch
+
+@Composable
+fun InfoSheet(
+    type: InfoSheetType?,
+    onClose: () -> Unit
+) {
+    val uriHandler = LocalUriHandler.current
+    if (type != null) {
+        when (type) {
+            is InfoSheetType.NetworkFeeSheetInfo -> NetworkFeeInfoSheet(
+                info = type,
+                onAction = { type.infoUrl?.let { uriHandler.open(it) } },
+                onClose = onClose
+            )
+
+            is InfoSheetType.TransactionStateSheetInfo -> TransactionInfoSheet(
+                info = type,
+                onAction = { type.infoUrl?.let { uriHandler.open(it) } },
+                onClose = onClose
+            )
+
+            is InfoSheetType.WalletWatchSheetInfo -> WalletWatchInfoSheet(
+                onAction = { type.infoUrl?.let { uriHandler.open(it) } },
+                onClose = onClose
+            )
+
+            is InfoSheetType.LockTimeSheetInfo -> LockTimeInfoSheet(
+                info = type,
+                onAction = { type.infoUrl?.let { uriHandler.open(it) } },
+                onClose = onClose,
+            )
+        }
+    }
+}
+
+sealed class InfoSheetType {
```

Comment:
Also try simplify.
```kotlin
sealed clas InfoSheetType(
   val iconUrl: String,
   val badgeIcon: String,
   @StrRes val title: Int,
   @StrRes val description: Int,
   ....
) {

}```


And you no need this https://github.com/gemwalletcom/gem-android/pull/67/files#diff-099581e4d670a4e2629291d62e7363c37e4a9879ebf4fd2281e9311eab0ea050R121